definitions.h

```diff
@@ -223,16 +223,60 @@ class BufferImage : public Buffer2D<PIXEL>
 class Attributes
 {      
     public:
-        PIXEL color;
-        // Obligatory empty constructor
-        Attributes() {}
+        void * data;
+        int dLen; // Know how much data is in an array
+
+        // No need for an empty constructor as this constructor will
+        // be called when no parameters are provided
+        Attributes(void * data = nullptr, int dLen = -1) 
+        {
+            this->data = data;
+            this->dLen = dLen;
+        }
 
         // Needed by clipping (linearly interpolated Attributes between two others)
         Attributes(const Attributes & first, const Attributes & second, const double & valueBetween)
         {
             // Your code goes here when clipping is implemented
         }
-};	
+};
+
+/*******************************************************
+ * Contains values for RgbAttributes
+ ******************************************************/
+class RgbAttr : public Attributes
+{
+    public:
+        RgbAttr(double red = 0, double green = 0, double blue = 0) 
+        {
+            rgb[0] = red;
```

Comment:
I like the array idea. it makes storing your attributes cleaner. I might implement this in mine.